tinygrad/tensor.py

```diff
@@ -515,9 +515,7 @@ def batchnorm(self, weight:Optional[Tensor], bias:Optional[Tensor], mean:Tensor,
 
   def dropout(self, p=0.5) -> Tensor:
     if not Tensor.training: return self
-    # TODO: why is this going through numpy?
-    _mask: np.ndarray = np.asarray(Tensor._rng.binomial(1, 1.0-p, size=self.shape), dtype=np.float32)
-    return self * Tensor(_mask, requires_grad=False, device=self.device) * (1/(1.0 - p))
+    return self * Tensor(Tensor._rng.binomial(1, 1.0 - p, size=self.shape), requires_grad=False, device=self.device, dtype=self.dtype) * (1 / (1.0 - p))
```

Comment:
This is still numpy

Reply:
@geohot I see, so is this about Tensor._rng not being based on numpy. should I replace it with https://docs.python.org/3/library/random.html?

Reply:
No, it's about reducing calls to Tensor._rng and funneling them into the tinygrad randomness methods we already have.

Reply:
by tinygrad randomness methods, do you mean Tensor.rand, Tensor.randn, etc? don't they also depend on Tensor._rng

Reply:
@geohot changed it to use Tensor.rand